improver/psychrometric_calculations/condensation_trails.py

```diff
@@ -367,3 +304,152 @@ def process(self, *cubes: Union[Cube, CubeList]) -> Cube:
         )
 
         return return_cube
+
+
+class ContrailHeightExtractor(BasePlugin):
+    """
+    Plugin to extract contrail formation heights by category. It extracts the maximum or minimum
+    height where contrail formation is Non-persistent or Persistent.
+    """
+
+    def __init__(self, use_max=True):
+        """
+        Initialize the Class
+
+        Args:
+            use_max (bool, optional):
+                If True, extract maximum heights; if False, extract minimum heights.
+        """
```

Comment:
```suggestion
    def __init__(self, use_max: bool=True):
        """
        Initialize the Class

        Args:
            use_max:
                If True, extract maximum heights; if False, extract minimum heights.
        """
```
The type hint means you don't need to add the type directly into the docstring (sphinx will add this to the auto-generated documentation). This is also suggested by [the IMPROVER code style guide](https://improver.readthedocs.io/en/stable/Code-Style-Guide.html#docstrings).